lib/slack-ruby-bot/commands/base.rb

```diff
@@ -1,7 +1,69 @@
 module SlackRubyBot
   module Commands
+
+    # class methods to modify the Commands::Base class. All methods are related to
+    # providing before/after hooks to commands.
+    module CommandHooks
```

Comment:
The code in `ActiveSupport::Callbacks` appears similar in intention. It's 600+ lines of meta-magic. I looked at it for 30m and barely understood it. My approach is simpler though less powerful/feature-rich.